yt/utilities/linear_interpolators.py

```diff
@@ -1,11 +1,82 @@
+import abc
+
 import numpy as np
 
 import yt.utilities.lib.interpolators as lib
 from yt.funcs import mylog
 
 
-class UnilinearFieldInterpolator:
-    def __init__(self, table, boundaries, field_names, truncate=False):
+class _LinearInterpolator(abc.ABC):
+    _ndim: int
+    _dim_i_type = "int32"
```

Comment:
wow, I'm sure this is just a your refactor making this more obvious, but do you have any idea why we are *not* using the same dtype in all interpolators ?

Reply:
I do not know :) I meant to look more closely, maybe there's some reason at the cython level? Or...?

Reply:
Ok, looked back and the superficial reason is that the cython functions for the different interpolators have different int declarations in their call signature: `UnilinearlyInterpolate` and `BilinearlyInterpolate` are declared as `int32_t`, e.g., : 

```
def BilinearlyInterpolate(np.ndarray[np.float64_t, ndim=2] table,
                          np.ndarray[np.float64_t, ndim=1] x_vals,
                          np.ndarray[np.float64_t, ndim=1] y_vals,
                          np.ndarray[np.float64_t, ndim=1] x_bins,
                          np.ndarray[np.float64_t, ndim=1] y_bins,
                          np.ndarray[np.int32_t, ndim=1] x_is,
                          np.ndarray[np.int32_t, ndim=1] y_is,
                          np.ndarray[np.float64_t, ndim=1] output):
```

while `TrilinearlyInterpolate` and `QuadrilinearlyInterpolate` use `int_t`:

```
def TrilinearlyInterpolate(np.ndarray[np.float64_t, ndim=3] table,
                           np.ndarray[np.float64_t, ndim=1] x_vals,
                           np.ndarray[np.float64_t, ndim=1] y_vals,
                           np.ndarray[np.float64_t, ndim=1] z_vals,
                           np.ndarray[np.float64_t, ndim=1] x_bins,
                           np.ndarray[np.float64_t, ndim=1] y_bins,
                           np.ndarray[np.float64_t, ndim=1] z_bins,
                           np.ndarray[np.int_t, ndim=1] x_is,
                           np.ndarray[np.int_t, ndim=1] y_is,
                           np.ndarray[np.int_t, ndim=1] z_is,
                           np.ndarray[np.float64_t, ndim=1] output):
```

As to **why** they differ, I'm really not sure... 

I could try switching `UnilinearlyInterpolate` and `BilinearlyInterpolate` over to use `int_t` as well?

Reply:
> I could try switching UnilinearlyInterpolate and BilinearlyInterpolate over to use int_t as well?

Now might be a good time to try indeed, but it's not critical !

Reply:
Just switched all the interpolators to use `int_t`